e2e/utils/ProfileUtils.ts

```diff
@@ -0,0 +1,339 @@
+// Deployment-profile-related utilities
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import util from 'util';
+
+import { expect, Page } from '@playwright/test';
+
+import {
+  createDefaultSettings, createUserProfile, startRancherDesktop, retry, teardown,
+} from './TestUtils';
+import { NavPage } from '../pages/nav-page';
+
+import * as childProcess from '@pkg/utils/childProcess';
+import paths from '@pkg/utils/paths';
+
+export async function clearSettings(): Promise<void> {
+  const fullPath = path.join(paths.config, 'settings.json');
+
+  await fs.promises.rm(fullPath, { force: true });
+}
+
+export async function clearUserProfile(): Promise<void> {
+  const platform = os.platform() as 'win32' | 'darwin' | 'linux';
+
+  if (platform === 'win32') {
+    return await verifyNoRegistrySubtree('HKCU');
+  }
+  const profilePaths = getDeploymentPaths(platform, paths.deploymentProfileUser);
+
+  for (const fullPath of profilePaths) {
+    await fs.promises.rm(fullPath, { force: true });
+  }
+}
+
+async function fileExists(fullPath: string): Promise<boolean> {
+  try {
+    await fs.promises.access(fullPath);
+
+    return true;
+  } catch { }
+
+  return false;
+}
+
+function getDeploymentBaseNames(platform: 'linux'|'darwin'): string[] {
+  if (platform === 'linux') {
+    return ['rancher-desktop.defaults.json', 'rancher-desktop.locked.json'];
+  } else if (platform === 'darwin') {
+    return ['io.rancherdesktop.profile.defaults.plist', 'io.rancherdesktop.profile.locked.plist'];
+  } else {
+    throw new Error(`Unexpected platform ${ platform }`);
+  }
+}
+
+function getDeploymentPaths(platform: 'linux'|'darwin', profileDir: string): string[] {
+  let baseNames = getDeploymentBaseNames(platform);
+
+  if (platform === 'linux' && profileDir === paths.deploymentProfileSystem) {
+    // macOS profile base-names are the same in both directories
+    // linux ones change...
+    baseNames = baseNames.map(s => s.replace('rancher-desktop.', ''));
+  }
+
+  return baseNames.map(baseName => path.join(profileDir, baseName));
+}
+
+async function hasSystemRegistrySubtree(): Promise<boolean> {
+  for (const profileType of ['defaults', 'locked']) {
+    for (const variant of ['Policies\\Rancher Desktop', 'Rancher Desktop\\Profile']) {
+      try {
+        const { stdout } = await childProcess.spawnFile('reg',
+          ['query', `HKLM\\SOFTWARE\\${ variant }\\${ profileType }`],
+          { stdio: ['ignore', 'pipe', 'pipe'] });
+
+        if (stdout.length > 0) {
+          return true;
+        }
+      } catch { }
+    }
+  }
+
+  return false;
+}
+
+export async function verifySystemRegistrySubtree(): Promise<string[]> {
+  if (await hasSystemRegistrySubtree()) {
+    return [];
+  } else {
+    return [`Need to add registry subtree "HKLM\\SOFTWARE\\Policies\\Rancher Desktop\\<defaults or locked>"`];
+  }
+}
+
+export async function verifySettings(): Promise<void> {
```

Comment:
I still think the `verify` wording is misleading. If you look at a [dictionary definition of verify](https://www.thefreedictionary.com/verify), there is nothing there that implies a state change - it is all about checking for something. Using `ensure` (or something better?) rather than `verify` in the cases where the system may be changed would be clearer. I would make these changes:
- verifySettings -> ensureSettings
- verifyUserProfile -> ensureUserProfile
- verifyNoRegistrySubtree -> clearRegistrySubtree
- verifyNoSystemProfile -> clearSystemProfile

Reply:
We agreed to keep the names as they are.